src/v1/order-forms/order-forms.controller.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+import { OrderFormsController } from './order-forms.controller.js';
+import { OrderFormsService } from './order-forms.service.js';
+import {
+  IMPORT_MODULES,
+  clearDB,
+  getSynchronizeConnection,
+} from '../../../test/utils/utils.js';
+import { OrderFormsModule } from './order-forms.module.js';
+import { instanceToInstance } from 'class-transformer';
+import { Brackets, DataSource } from 'typeorm';
+import { User } from '../../entities/user.entity.js';
+import { Branch } from '../../entities/branch.entity.js';
+import { CourtInfo } from '../../entities/courtinfo.entity.js';
+import { INestApplication } from '@nestjs/common';
+import { UserRole } from '../../commons/enums/UserRole.enum.js';
+import { TimeSlot } from '../../entities/timeslot.entity.js';
+
+describe('OrderFormsController', () => {
+  let app: INestApplication;
+  let datasource: DataSource;
+  let catsService = { findAll: () => ['test'] };
+  beforeAll(async () => {
+    datasource = await getSynchronizeConnection();
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [...IMPORT_MODULES, OrderFormsModule],
+    }).compile();
+    app = module.createNestApplication();
+    await app.init();
+  });
+
+  afterEach(async () => {
+    // await clearDB(datasource);
+  });
+  describe('POST order-forms', () => {
+    let user: User, admin: User, court: CourtInfo, branch: Branch;
+    /*
+      Initialize necessary data to test order.
+      Initialize user, admin, court, branch.
+    */
+    beforeEach(async () => {
+      //initialize sender user
+      user = new User();
+      user.username = 'Josh';
+      user.email = 'orz@gmail.com';
+      user.password = 'dhruv';
+      user.fullname = 'Josh Johan';
+      user.phone = '0121224134';
+      await user.save();
+      console.log(user);
+      //initialize admin
+      admin = new User();
+      admin.username = 'admin';
+      admin.email = 'admin@badminton.com';
+      admin.password = 'dhruv';
+      admin.fullname = 'Son tran viet';
+      admin.role = UserRole.ADMIN;
+      admin.phone = '0796688018';
+      await admin.save();
+      //initialize branch
+      branch = new Branch();
+      branch.owner = admin;
+      branch.address = 'Le Dinh Ly, Da Nang';
+      await branch.save();
+      //initialize court
+      court = new CourtInfo();
+      court.branch = branch;
+      court.id_branch = branch.id;
+      court.description = 'Sân trong nhà, gần khán đài';
+      court.name_of_court = 'San 1';
+      await court.save();
+    });
```

Comment:
Move to test/ultil.ts

createRandomUser() function, using `@faker-js/faker` for random data

Reply:
ex:

```js
const { faker } = require('@faker-js/faker');

function getRandomUser(role = UserRole.USER) {
  const user = new User();
  user.username = faker.internet.userName();
  user.email = faker.internet.email();
  user.password = faker.internet.password();
  user.fullname = faker.name.fullName();
  user.phone = faker.phone.number();
  user.role = role;
  return user;
}
```

Reply:
```
const { user, admin, branch, court } = await initCourtData();
```